lib/sphinx_rtm/messages.ex

```diff
@@ -0,0 +1,66 @@
+defmodule SphinxRtm.Messages do
+  use Slack
+  alias Slack.Web.Chat
+  alias Slack.Web.Users
+  alias Sphinx.Riddles
+
+  ## TODO: check if message is a question
+  ## if yes then get the keywords and search for old questions
+  ## else check if it is related to/ answering the recent question.
+  ## If yes, save the answer
+  ## Else discard (something like that?)
+  @spec process(map()) :: {:ok, Riddles.Riddle.t()} | {:error, Ecto.Changeset.t()}
+  def process(message) do
+    case Map.has_key?(message, :thread_ts) do
+      true ->
+        process_reply(message)
+
+      false ->
+        # TODO: check if @sphinx is invoked
+        process_question(message)
+    end
+  end
+
+  @spec process_question(map()) :: {:ok, Riddles.Riddle.t()} | {:error, Ecto.Changeset.t()}
+  defp process_question(message) do
+    %{}
+    |> Map.put(:enquirer, get_user_name(message.user))
+    |> Map.put(:title, message.text)
+    |> Map.put(:permalink, get_permalink(message.channel, message.ts))
+    |> Riddles.create()
+  end
+
+  @spec process_reply(map()) :: {:ok, Riddles.Riddle.t()} | {:error, Ecto.Changeset.t()}
+  defp process_reply(message) do
+    id =
+      get_permalink(message.channel, message.thread_ts)
+      |> get_riddle_id()
+
+    %{}
+    |> Map.put(:solver, get_user_name(message.user))
+    |> Map.put(:permalink_answer, get_permalink(message.channel, message.ts))
+    |> Map.put(:id, id)
+    |> Riddles.update()
+  end
+
+  @spec get_permalink(String.t(), String.t()) :: String.t()
```

Comment:
@filipevarjao I'm thinking those `get_permalink/2` and `get_user_name/1` are quite generic and can be moved to a "library" type of module. What do you think? Though I have not think of any good name for it yet 😛

Reply:
yeah, sure. Maybe riddles_support, riddles_utils or so 
or even move it to Riddles

Reply:
I think they are more a Slack related function though 🤔 But slack_utils maybe?

Reply:
yes